services/diagnostic/api.go

```diff
@@ -110,7 +110,6 @@ func (s *SessionService) handleSessions(w http.ResponseWriter, r *http.Request)
 	header := w.Header()
 	header.Add("Transfer-Encoding", "chunked")
 	header.Add("Content-Type", r.Header.Get("Content-Type"))
-	w.WriteHeader(http.StatusOK)
```

Comment:
Hmm, how was this getting called multiple times?

Reply:
I'm trying to find the exact code, but there's a race between the call to `w.WriteHeader` and the first call to `w.Flush` which checks internally whether headers have been written already or not.

My thoughts were to let the internal HTTP logic handle this instead of writing the header ourselves. Thoughts?

Reply:
If there is a race with `w.Flush` then there is likely a race with `w.Header()`. We need to fix the race, which probably just means moving lines 107-108 down after anything that interacts with `w`.